<launch>

  <arg name="robot" default="mynteye"/>
  <arg name="fixed_frame_id" default="mynteye_link"/>
  <arg name="calibration_file"
    default="$(find msckf_vio)/config/camchain-imucam-mynteye-s1030.yaml"/>
  <arg name="rviz" default="false"/>

  <group if="$(arg rviz)">
      <node pkg="rviz" type="rviz" name="rviz" args="-d $(find msckf_vio)/rviz/rviz_mynteye_config.rviz"/>
  </group>

  <!-- Image Processor Nodelet  -->
  <include file="$(find msckf_vio)/launch/image_processor_mynteye.launch">
    <arg name="robot" value="$(arg robot)"/>
    <arg name="calibration_file" value="$(arg calibration_file)"/>
  </include>

  <!-- Msckf Vio Nodelet  -->
  <group ns="$(arg robot)">
    <node pkg="nodelet" type="nodelet" name="vio"
      args='standalone msckf_vio/MsckfVioNodelet'
      output="screen">

      <!-- Calibration parameters -->
      <rosparam command="load" file="$(arg calibration_file)"/>

      <param name="publish_tf" value="true"/>
      <param name="frame_rate" value="10"/>
      <param name="fixed_frame_id" value="$(arg fixed_frame_id)"/>
      <param name="child_frame_id" value="odom"/>
      <param name="max_cam_state_size" value="20"/>
      <param name="position_std_threshold" value="8.0"/>

      <param name="rotation_threshold" value="0.2618"/>
      <param name="translation_threshold" value="0.4"/>
      <param name="tracking_rate_threshold" value="0.5"/>

      <!-- Feature optimization config -->
      <param name="feature/config/translation_threshold" value="-1.0"/>

      <!-- These values should be standard deviation -->
      <param name="noise/gyro" value="0.00888232829671"/>
      <param name="noise/acc" value="0.0268014618074"/>
      <param name="noise/gyro_bias" value="0.000379565782927"/>
      <param name="noise/acc_bias" value="0.00262960861593"/>
      <param name="noise/feature" value="0.035"/>

      <param name="initial_state/velocity/x" value="0.0"/>
      <param name="initial_state/velocity/y" value="0.0"/>
      <param name="initial_state/velocity/z" value="0.0"/>

      <!-- These values should be covariance -->
      <param name="initial_covariance/velocity" value="0.25"/>
      <param name="initial_covariance/gyro_bias" value="0.01"/>
      <param name="initial_covariance/acc_bias" value="0.01"/>
      <param name="initial_covariance/extrinsic_rotation_cov" value="3.0462e-4"/>
      <param name="initial_covariance/extrinsic_translation_cov" value="2.5e-5"/>

      <remap from="~imu" to="/mynteye/imu/data_raw"/>
      <remap from="~features" to="image_processor/features"/>

    </node>
  </group>

</launch>
